### ISSUE TITLE
document API

### ISSUE THREAD

**Issue by NickleDave** [sepandhaghighi/pycm#16]
Hi @sepandhaghighi I'm reviewing `pycm` for JOSS.
First, you all have done a great job and the library is well put-together. I'm sorry I haven't started the review sooner. I think the library as-is is very close to "accept". There are just a couple of minor revisions I think might be helpful.

Of those minor revisions, the one I would most like to see is a thorough documentation of the API.

At the bare minimum, the user should be able to type `help(ConfusionMatrix)` (or `ConfusionMatrix?` in Ipython) and get some sort of useful help. As you probably know, the way to fix this would be to add a docstring to the `ConfusionMatrix` class. This docstring should state what the acceptable types for `y_true` and `y_pred` are--are only `lists` of `ints` and `np.ndarrays` of `ints` valid? Including more brief versions of the examples from the README.md (without the entire output of the `print(cm)` statement) would probably also be helpful.

I think it would also be helpful to state what data types are acceptable in plain English on your README.md. A potential user will want to know if `pymc` can accept their data in its present state or if they're going to have to convert it somehow to the right format.

I will need to finish this review tomorrow but I wanted to get it started for you as soon as possible. I hope my review will be helpful and we can quickly get your library into JOSS so more people can appreciate it.

**Comment by sepandhaghighi** [comment]
Hi @NickleDave 

Thanks for your point, I will fix this in next few hours.
